lib/hanami/cli/generator.rb

```diff
@@ -0,0 +1,146 @@
+require 'hanami/utils/files'
+require 'erb'
+
+module Hanami
+  class CLI
+    # A class to help generate files
+    #
+    # @since 0.1.0
+    class Generator
+      # @since x.x.x
+      # @api public
+      def initialize(out: $stdout, files: Utils::Files, templates_dir: __dir__)
+        @files         = files
+        @out           = out
+        @templates_dir = Pathname.new(templates_dir)
+      end
+
+      # @since x.x.x
+      # @api public
+      def create(source, destination, context)
+        files.write(destination, render(source, context))
+        say(:create, destination)
+      end
+
+      # @since x.x.x
+      # @api public
+      def touch(path)
+        files.touch(path)
+        say(:create, path)
+      end
+
+      # @since x.x.x
+      # @api public
+      def copy(source, destination)
+        files.cp(template_path(source), destination)
+        say(:create, destination)
+      end
+
+      # @since x.x.x
+      # @api public
+      def delete(path, allow_missing: false)
+        return if allow_missing && !files.exist?(path)
+
+        files.delete(path)
+        say(:remove, path)
+      end
+
+      # @since x.x.x
+      # @api public
+      def delete_directory(path, allow_missing: false)
+        return if allow_missing && !files.exist?(path)
+
+        files.delete_directory(path)
+        say(:remove, path)
+      end
+
+      # @since x.x.x
+      # @api public
+      def remove_line(path, content)
+        files.remove_line(path, content)
+        say(:subtract, path)
+      end
+
+      # @since x.x.x
+      # @api public
+      def insert_after_first(path, content, after:)
+        files.inject_line_after(path, after, content)
+        say(:insert, path)
+      end
+
+      # @since x.x.x
+      # @api public
+      def insert_after_last(path, content, after:)
+        files.inject_line_after_last(path, after, content)
+        say(:insert, path)
+      end
+
+      # @since x.x.x
+      # @api public
+      def append(path, content)
+        files.append(path, content)
+        say(:append, path)
+      end
+
+      # @since x.x.x
+      # @api public
+      def execute(command, **opts)
+        system(command, opts)
+        say(:run, command)
+      end
+
+      private
+
+      # @since x.x.x
+      # @api private
+      SAY_FORMATTER = "%<operation>12s  %<path>s\n".freeze
+
+      # @since x.x.x
+      # @api private
+      class Renderer
```

Comment:
I'm not sure we need this class? Seems like it could just be a helper method.

It was taken from: https://github.com/hanami/hanami/blob/master/lib/hanami/cli/commands/command.rb#L105-L125